openml-lightgbm/lightgbm-provider/src/main/java/com/feedzai/openml/provider/lightgbm/LightGBMUtils.java

```diff
@@ -82,6 +84,31 @@ static CpuArchitecture getCpuArchitecture(final String cpuArchName) {
         }
     }
 
+    /**
+     * Checks if the current operative system is Alpine.
+     *
+     * @return true when the operative system is Alpine and false otherwise.
+     */
+    static private boolean isAlpine() {
+        final String osReleasePath = "/etc/os-release";
+
+        try {
+            BufferedReader reader = new BufferedReader(new FileReader(osReleasePath));
+
+            String line;
+            while ((line = reader.readLine()) != null) {
+                if (line.equals("ID=alpine")) {
+                    reader.close();
```

Comment:
You don't need to close the reader if you initialize it in a try-with-resources (https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html).

Reply:
This code was removed in the latest commits, but I will keep that in mind.